compiler/rustc_builtin_macros/src/unreachable.rs

```diff
@@ -0,0 +1,63 @@
+use rustc_ast::ptr::P;
+use rustc_ast::tokenstream::{DelimSpan, TokenStream};
+use rustc_ast::*;
+use rustc_expand::base::*;
+use rustc_span::edition::Edition;
+use rustc_span::symbol::sym;
+use rustc_span::Span;
+
+// This expands to either
+// - `$crate::panic::unreachable_2015!(...)` or
+// - `$crate::panic::unreachable_2021!(...)`
+// depending on the edition.
+//
+// `$crate` will refer to either the `std` or `core` crate depending on which
+// one we're expanding from.
```

Comment:
This part is not true for `unreachable`, as `std::unreachable` is just a re-export of `core::unreachable`. So `$crate` will always be `core` here.